sdk/communication/CONTRIBUTING.md

```diff
@@ -0,0 +1,76 @@
+
+When you go inside the package you are working with, you will see a folders called `recordings\browsers` and `recordings\node`. In JS we performs tests in the node and browser environment. Each folder contains, as its name suggests, recordings of successful calls to the API that allow us to run the tests in PLAYBACK mode and remove the necessity of hitting the actual resources every time we may want to test.
+
+To test with environment variables, JS has a `sample.env` for developers to use as a template. You can rename the file to `.env` and add the appropriate environment variables.
+
+### Playback mode
+
+By default tests are run in playback mode. You can also set an environment variable in your `.env` called `TEST_MODE` and set its value to `PLAYBACK`. You can then run `rushx test` command to run the tests.
+
+If the tests are successful, we can proceed to run the tests in LIVE mode.
+
+### Live mode
+
+Since in LIVE mode we are hitting an actual resource, we must set the appropriate environment variables to make sure the code tests against the resource we want. Set up an env variable called `AZURE_COMMUNICATION_LIVETEST_CONNECTION_STRING` for Phone Number and SMS packages and set it to the connection string of the resource you want to test against. For all other communication packages, set up an environment variable called `COMMUNICATION_CONNECTION_STRING` and set it to the connection string of the resource you want to test against.
+
+Depending on which package you are testing, it may need special environment variables to test succesfully. In each package, there is a `recordedClient.ts` file; In that file you will find `const replaceableVariables` with the names of the variables the package uses. Make sure to set these variables before running the tests themselves.
+
+After setting up the environment variables and setting `TEST_MODE` to `LIVE`, you can run the tests in the Test Explorer.
+
+### Record mode
+
+RECORD mode is similar to LIVE mode because it also hits an actual resource. In addition to hitting the resource, RECORD mode will also record the successful calls to the service in json format. As mentioned, the recordings are stored in the test package under `recordings\browsers` and `recordings\node`.
+
+If you would like to generate new recordings for a test, setting `TEST_MODE` to `RECORD`.
+
+### Managed Identity Tests
+
+If you ran the tests in RECORD mode, you may have noticed that the files inside the `recordings` folder were updated. If any of the tests failed, you will see the error message in your terminal logs.
+
+The most probable thing is that the managed identity tests will fail at first. This is because we haven't set up any managed identity credentials for the `DefaultAzureCredential` object inside the tests to reference to. Managed identity tests do not execute in `browser` mode because of a security limitation using `DefaultAzureCredential`.
+
```

Comment:
@beltr0n I know you are working on a fix for this user the SecretCredential. So after that is merged in we will need to update this point